cities-debug

abotas opened the pull request:
S2Si sets its peaks to be those present in s2d. This is unacceptable when the sipms do not have signal during the one or more of the peaks in s2d. this has been fixed by `check_s2d_and_s2sid_share_peaks` in `pmaps.pyx`.

Commit:
write and test func to ensure s2d, s2sid have same peaks

s2si automatically set its peaks to be the peaks in s2d even if the
s2sid did not have all those peaks. this has been fixed

jjgomezcadenas reviewed (dismissed):
This was indeed a needed fix (which indeed reveals a strong feature of the new interfaces, eg., the capability to ensure consistency between peaks and sipms). The code is clear and is exercised by a test.

Commit:
fixes bug in find_s12 that could poorly initialize candidate peak

find_s12 sometimes initialized the start of the first peak to the
incorrect time. this has been fixed and a test has been added.

Commit:
index into KrEvent lists with enumeration of peak_no not peak_no

Commit:
make sure cython rebin_array returns nparray not memview

Commit:
address unshared peaks in rebin_s2si_peak

Commit:
stop penthesilea from printing garbage

a function in pmap_functions_c was printing a few integers for each
event

Commit:
move load_dst to io.dst_io from reco.dst_functions

jmbenlloch reviewed (approved):
This PR fixes some issues with peak numbers in several places and a bug in find_s12.
